### PR TITLE
Add planning and execution for real robot

### PR DESCRIPTION
**Pull request by tdl-ua**
Added code for planning and execution of trajectories on a real robot.
Code was tested on a sia5f robot arm. Planned trajectories were executed.

### CODE REVIEW

**Comment by Danfoa** [comment]
Hi @tdl-tbslab, I am also integrating one motoman robot with ros for real hardware operation but I do have a question, I hope you can help me with it.

How and where is defined the ordering of the custom joint names of the `controller_joint_names` parameter?; for example I am using a dual arm robot and I am not sure whether this order is correct.

```
controller_joint_names: ['arm_left_joint_1_s','arm_left_joint_2_l','arm_left_joint_3_e'
,'arm_left_joint_4_u','arm_left_joint_5_r','arm_left_joint_6_b','arm_left_joint_7_t',
'arm_right_joint_1_s','arm_right_joint_2_l','arm_right_joint_3_e','arm_right_joint_4_u',
'arm_right_joint_5_r','arm_right_joint_6_b','arm_right_joint_7_t','torso_joint_b1','torso_joint_b2']
```

Sorry to bother mate, hope you can clear this up

I created a ROS-Answers question if anyone wants to answer [there](https://answers.ros.org/question/288716/controller_joint_names-param-joint-order-for-motoman-driver-nodes/)

File: motoman_sia5d_moveit_config/launch/moveit_planning_execution.launch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  <!-- run the robot simulator and action interface nodes -->
+  <group if="$(arg sim)">
+    <include file="$(find industrial_robot_simulator)/launch/robot_interface_simulator.launch" />
+  </group>
+
+  <!-- run the "real robot" interface nodes -->
+  <!--   - this typically includes: robot_state, motion_interface, and joint_trajectory_action nodes -->
+  <!--   - replace these calls with appropriate robot-specific calls or launch files -->
+  <group unless="$(arg sim)">
+    <node pkg="rosservice" type="rosservice" name="robot_enable" args="call --wait /robot_enable"/>
```
Comment (gavanderhoorn) [inline motoman_sia5d_moveit_config/launch/moveit_planning_execution.launch:33]: @shaun-edwards: is this something we want to do -- automatically -- in a launch file?

I understand that this is convenient, but 'safety' never is convenient.

Comment (tdl-ua) [reply motoman_sia5d_moveit_config/launch/moveit_planning_execution.launch:33]: That's a good point. It is probably safer for the user to handle this separately. I can remove the line and create another pull request.

**Comment by tdl-ua** [comment]
Hi @Danfoa, 

My knowledge of ros-industrial and related packages is not too deep (yet). I stuck with [this](http://wiki.ros.org/Industrial/Tutorials/Create_a_MoveIt_Pkg_for_an_Industrial_Robot) tutorial for putting together the code. In step 2.2, the _controller_joint_names_ parameter is mentioned, which is required when a robot has non-standard joint names. In there it says "The order of this list should match that expected by your robot's socket interface. Typically Base -> Tip." Following this instruction and the tags on our sia5f robot, the joint order would be [S L E U R B T]. This would also agree with the description in section 3.2 of [wiki.ros.org/industrial_robot_client/joint_naming](http://wiki.ros.org/industrial_robot_client/joint_naming). Looking at the _joint_names_sia5d.yaml_-file in _motoman/motoman_sia5d_support/config/_, the same joint order is found.

I am, however, not familiar with dual arm configurations. I can only see one typo in your order. I think the second `arm_right_joint_1_s` should be named 'arm_right_joint_2_s'.

**Comment by Danfoa** [comment]
Thank you very much for the help @tdl-tbslab :).
I will post here or in the question link if I have any updates.

**Pull request merged**